### PR TITLE
`Development`: Reorganize setup documentation

### LINKED ISSUE #5657: `Development`: Follow-up: Split Setup documentation into a developer and production part
### Is your feature request related to a problem?

In #5648 we came to the conclusion that it would be beneficial to separate the developer and production setup documentation.
Why? Because currently it's quite a big setup page with a lot of information the usual developer/contributer doesn't need to know to get started for development.

### Describe the solution you'd like

It was proposed to move the production setup documentation  to the `Administration Guide / Production Setup Guide`?

The development setup documentation can stay at `Contributor Guide / Setup Guide`.

This should happen in a way that the production setup documentation references to the developer setup documentation for basic setup steps, but goes further in detail on how to secure it and has more details about features like deploying multiple Artemis instances, ...

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #6449: `Documentation`: Move server customisation to Admin docs
### Is your feature request related to a problem?

The server customisation section ‘Customize your Artemis instance’ is currently part of the development setup documentation.

### Describe the solution you'd like

I think it should be moved to the administration section instead, since the imprint etc. is not relevant for dev setups but important for production ones.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context
Closes #6449.
Closes #5657.

The setup documentation at the moment interleaves documentation that is only relevant for local development setups and documentation that is only relevant for production setups. This makes it hard to follow the documentation in both cases.

### Description
This PR aims to reorganise the sections of the setup documentation so that ones only relevant for production setups (e.g., customisation, privacy statement) are moved to the admin docs.
Additionally, the setup for extension services is also moved to the admin docs. To make sure users are aware of those special services, a note is added to the start of the development docs that explains that this page only explains the core setup but the additional services to extend Artemis functionality are available.

Thus, the actual content of the individual sections has been mostly unchanged (except for small formatting adjustments like adapting the section nesting depth) and only been extracted into separate files and moved into a different directory.

As a future PR the Gitlab/Jenkins setup docs can be further improved to move the nginx-setup instructions from there to the admin section as well. However, in this first step this has not been done, since it would require restructuring within the section, rather than moving entire independent sections from dev to admin docs.

### Steps for Testing
Look at the setup docs for developers and admins and check that the sections you would expect are still there and that images are shown.

- https://artemis-platform--7323.org.readthedocs.build/en/7323/dev/setup.html
- https://artemis-platform--7323.org.readthedocs.build/en/7323/admin/setup.html
- https://artemis-platform--7323.org.readthedocs.build/en/7323/admin/extension-services.html

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### CODE REVIEW

File: docs/dev/setup/server.rst [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+These events are tracked by attaching a POST call to the respective DOM elements on the client side.
+The POST call accesses the **TextAssessmentEventResource** which then adds the events in its respective table.
+This feature is disabled by default. We can enable it by modifying the configuration in the file:
+``src/main/resources/config/application-artemis.yml`` like so:
+
+.. code:: yaml
+
+   info:
+      text-assessment-analytics-enabled: true
```
Comment (DominikRemo) [inline docs/dev/setup/server.rst:291]: This was moved to docs/admin/setup/text-assessment-analytics.rst

We should remove this occurrence

File: docs/admin/setup/legal/privacy-statement-templates.rst [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Privacy Statement German template
-"""""""""""""""""""""""""""""""""
-.. code-block:: none
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: markdown
```
Comment (DominikRemo) [inline docs/admin/setup/legal/privacy-statement-templates.rst:4]: Before these Changes, the templates were not highlighted at all. Now, the German privacy statement (top of image) is highlighted differently compared to the English version (bottom of image) and the imprint templates (headings not highlighted):
Screenshot 2023-10-16 at 16 17 24

File: docs/admin/setup/kubernetes.rst
Comment (DominikRemo) [inline docs/admin/setup/kubernetes.rst:1]: IMO, the Kubernetes guide should stay in the dev setup guide because "this does not result in a production-ready deployment"

**Review by DominikRemo — CHANGES_REQUESTED** [review]
Already looks good. Left some comments

File: docs/admin/setup/kubernetes.rst
Comment (b-fein) [reply docs/admin/setup/kubernetes.rst:1]: Neither does it result in a working dev-deployment at the moment. I’d leave it in the admin section as a special deployment similar to the multi-instance Artemis setup. Maybe someone that is familiar with Kubernetes can extract relevant pointers about how to deploy Artemis in their cluster using this documentation whereas I don’t think any developer is going to set up their dev-env using Kubernetes.

I think the goal is to whenever https://github.com/ls1intum/artemis-helm is ready, completely replace this part of the setup documentation with an updated one anyway.

**Review by DominikRemo — APPROVED** [review]
Thank you for implementing my feedback and improving our documentation :)

File: docs/admin/setup/kubernetes.rst [hunk 3]
```diff
@@ -1,11 +1,19 @@
...
------------------------------
+Kubernetes Setup
+----------------
 
 This section describes how to set up an environment deployed in Kubernetes.
 
+.. note::
+
+   These steps were part of an experimental setup.
+   This does not result in a production-ready deployment.
```
Comment (laadvo) [inline docs/admin/setup/kubernetes.rst:9]:
```suggestion
   These steps were part of an experimental setup and do not result in a production-ready deployment.
```

**Review by laadvo — APPROVED** [review]
The new reorganized setup documentation is easier to follow, as only steps that are relevant are provided and others are moved to the admin section. LGTM, just left one optional suggestion.
